sky/utils/tpu_utils.py

```diff
@@ -0,0 +1,25 @@
+"""Utility functions for TPUs."""
+from typing import Optional
+
+from sky import resources as resources_lib
+
+
+def is_tpu_vm(resources: resources_lib.Resources) -> bool:
+    if resources.accelerator_args is None:
+        return False
+    return resources.accelerator_args.get('tpu_vm', False)
+
+
+def is_tpu_pod(resources: resources_lib.Resources) -> bool:
+    if resources.accelerators is None:
+        return False
+    acc, _ = list(resources.accelerators.items())[0]
+    return acc.startswith('tpu') and acc not in ['tpu-v2-8', 'tpu-v3-8']
+
+
+def get_num_tpu_devices(resources: resources_lib.Resources) -> Optional[bool]:
+    if not is_tpu_vm(resources):
```

Comment:
nit: Maybe we can change it to `if not acc.startswith('tpu')` to use this function later for TPU Nodes?

Reply:
If you change it, please also change `cloud_vm_ray_backend.py:L2923`

Reply:
@infwinston I meant https://github.com/skypilot-org/skypilot/blob/3dd2b2a05ed3f11e6e8da4671980928253904220/sky/backends/cloud_vm_ray_backend.py#L2134 we should additionally check `is_tpu_vm` in this line, shouldn't we?

Reply:
good point. I fixed it with some slight function refactoring.